### PR TITLE
Change DIIS in MRDSRG

### PR DESCRIPTION
**Pull request by lcyyork**
## Description
- [x] Avoid storing a big amplitudes vector for DIIS extrapolation in `MRDSRG` class. And thus it avoids two copies between the `vector` and `ambit::Tensor`.
- [x] Changed `MRDSRG` amplitudes update, where `Hbar` is used as an intermediate. However, the part of `Hbar` (used for reference relaxation) is now kept unchanged.
- [x] Clean up the `mrdsrg_pt.cc` file for `MRDSRG`.
- [x] Remove `boost/format.hpp` in `mrdsrg_pt.cc` and `mrdsrg_nonpt.cc`.

## Notes
- [x] This PR only affects the spin-integrated `MRDSRG` class.
- [x] `DIISManager` now sets to `OnDisk`.
- [x] Add a new file: `mrdsrg_diis.cc`.
- [x] Documented keywords for DSRG DIIS controls.
- [x] Documented different zeroth-order Hamiltonians used for testing DSRG PT2s.

## Checklist
- [x] Add a test case: `mrdsrg-pt2-5`, running time: ~100s
- [x] Removed comments in code and input files
- [ ] Documented source code (How to do this?)
- [x] Checked for redundant headers
- [x] Checked for consistency in the formatting of the output file
- [x] Documented new features in the manual
- [x] Ready to go!

### CODE REVIEW

**Comment by fevangelista** [comment]
@lcyyork: thanks for the update. Does this PR add DIIS to all implementations of the MR-DSRG? Would be interesting to see any difference in timings if there are any.

**Comment by lcyyork** [comment]
@fevangelista No, it only affects the spin-integrated MRDSRG class. I did not observe any large time changes because the test case is very small. This avoid two copies of the hhpp part, which is a minor effect for the total timing/storage.

I also change the amplitude update such that Hbar does not need to be recomputed before reference relaxation. This can save a Hbar building time.

**Comment by fevangelista** [comment]
@lcyyork : this looks ready to be merged. Can you please run all the mr-dsrg test cases, not just the ones on TravisCI? Just to check.

File: src/mrdsrg-spin-integrated/mrdsrg_diis.cc [hunk 1]
```diff
@@ -0,0 +1,209 @@
...
+            amp_ptrs_.push_back(T1_.block(block).data().data());
+            res_ptrs_.push_back(DT1_.block(block).data().data());
+        } else {
+            sizes[i] = T2_.block(block).numel();
+            amp_ptrs_.push_back(T2_.block(block).data().data());
+            res_ptrs_.push_back(DT2_.block(block).data().data());
+        }
+    }
+
+    diis_manager_->set_error_vector_size(
```
[the remark this answers is not in the document]
Comment (lcyyork) [reply src/mrdsrg-spin-integrated/mrdsrg_diis.cc]: Yes, unfortunately this is the C variadic style.

File: src/mrdsrg-spin-integrated/dsrg_mrpt2.cc [hunk 2]
```diff
@@ -85,14 +85,14 @@ void DSRG_MRPT2::startup() {
...
-        relax_ref_ = "ONCE";
+    //    if (relax_ref_ != "NONE" && relax_ref_ != "ONCE") {
+    //        outfile->Printf("\n\n  Warning: RELAX_REF option \"%s\" is not supported. Change to
+    //        ONCE.", relax_ref_.c_str());
+    //        relax_ref_ = "ONCE";
 
-        warnings_.push_back(std::make_tuple("Unsupported RELAX_REF", "Change to ONCE",
-                                            "Change options in input.dat"));
-    }
+    //        warnings_.push_back(std::make_tuple("Unsupported RELAX_REF", "Change to ONCE",
```
[the remark this answers is not in the document]
Comment (lcyyork) [reply src/mrdsrg-spin-integrated/dsrg_mrpt2.cc]: OK.

**Comment by lcyyork** [comment]
@fevangelista OK, I will run the test cases.

**Pull request merged**